android/app/src/main/java/com/rjnr/pocketnode/data/wallet/WalletPreferences.kt

```
@@ -19,81 +21,139 @@ class WalletPreferences @Inject constructor(
-        val modeName = prefs.getString(KEY_SYNC_MODE, SyncMode.RECENT.name)
+    init {
+        migrateIfNeeded()
+    }
+
+    // --- Network selection (global, not namespaced) ---
+
+    fun getSelectedNetwork(): NetworkType {
+        val name = prefs.getString(KEY_SELECTED_NETWORK, NetworkType.MAINNET.name)
+        return try {
+            NetworkType.valueOf(name ?: NetworkType.MAINNET.name)
+        } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)